tests/test_camera.py

```diff
@@ -25,12 +25,13 @@ class CameraTest(unittest.TestCase):
 
     # TODO add test for Serial number
     def setUp(self):
-        mock = MockRobotD(root_dir="/tmp/robotd")
+        self.root_dir = create_root_dir()
```

Comment:
There's a lot of repetition here, perhaps we could extract a mixin which handles all of this?

```python
class MockRobotDMixin:
    def create_robotd_mock(self):
        root_dir = tempfile.TemporaryDirectory(prefix="robot-api-test-")
        self.addCleanup(root_dir.cleanup)
        mock_robotd = MockRobotD(root_dir=root_dir.name)
        self.addCleanup(mock_robotd.stop)
        return mock_robotd
```
Usage looks like this:
```python
class MyTests(MockRobotDMixin, unittest.TestCase):
    def setUp(self):
        self.mock_robotd = self.create_robotd_mock()
        ...
        self.robot = Robot(robotd_path=self.mock_robotd.root_dir)
    ...
```

Reply:
I like this - implemented in 8275d72.